src/coreclr/jit/lclvars.cpp

```diff
@@ -2595,25 +2595,35 @@ void Compiler::lvaSetVarLiveInOutOfHandler(unsigned varNum)
 void Compiler::lvaSetVarDoNotEnregister(unsigned varNum DEBUGARG(DoNotEnregisterReason reason))
 {
     noway_assert(varNum < lvaCount);
-    LclVarDsc* varDsc         = &lvaTable[varNum];
+    LclVarDsc* varDsc = &lvaTable[varNum];
+    if (varDsc->lvDoNotEnregister == 1)
+    {
+        return;
```

Comment:
Is it reasonable to add `assert(varDsc->m_doNotEnregReason == reason)`?

Reply:
Or if `varDsc` was set to DoNotRegister by some other phase, then also set the reason here?

Reply:
I was thinking to do `DoNotEnregisterReason` enum to be a set like some other enums:
```
#define MAKE_DNRREASON(bit) (1ULL << (bit))
AddrExposed= = MAKE_DNRREASON( 0),
DontEnregStructs= = MAKE_DNRREASON( 1),
```
but because one reason is often causing others for the purpose of this dump it is clearer to keep only the first reason and not all of them for each var.

So we record only the first reason why something can't be put into a register and if different reasons come we ignore it.